astrogrism/simulate/simulate.py

```diff
@@ -0,0 +1,86 @@
+from os import environ
+from pathlib import Path
+from shutil import copy
+from tarfile import open as tar_open
+from tempfile import gettempdir
+from warnings import warn
+
+from astropy.utils.data import download_file
+from synphot import Observation
+
+
+def generate_simulation_spectrum(grism, detector=None, temp_path=gettempdir()):
+    """
+    Initializes and uses STSynphot to generate a Vega spectrum within the bandpass of a given grism
+
+    Parameters
+    ----------
+    grism : str
+        String representation of one of the four supported HST Grisms
+        Valid grisms: G141, G102, G280, G800L
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+
+    temp_path : str
+        Path to download necessary files for STSynphot. Fallsback to Python's
+        default temporary folder location
+
+    """
+    SIM_DATA_DIR = Path(temp_path) / "astrogrism_simulation_files"
+
+    if detector not in (1, 2, None):
+        raise ValueError("Invalid detector argument. Please choose 1 or 2")
+
+    # Prepare environment required for stsynphot to be imported
+    environ['PYSYN_CDBS'] = str(SIM_DATA_DIR / 'grp' / 'redcat' / 'trds')
+
+    # Always download our data files fresh. Ensures changes from redcat team will be picked up
+    # Download HST Instrument Data Files
+    SIM_DATA_DIR.mkdir(parents=True, exist_ok=True)
+    hst_data_files_archive = Path(download_file('https://ssb.stsci.edu/trds/tarfiles/synphot1.tar.gz', cache=True)) # noqa
+    with tar_open(hst_data_files_archive) as tar:
+        tar.extractall(path=SIM_DATA_DIR)
```

Comment:
Another thought here is... Since you only need just a few bandpasses, I wonder if you can precompute the final bandpass from obsmode outside of this package, and then ship the bandpass files as package data. Since you want to use `Spectrum1D` anyway, maybe you can use `specutils` to convolve Vega and bandpass (do they support that, @rosteen?) and not have synphot as dependency at all.

Reply:
Wouldn't we still need `synphot` to actually read in those bandpass files? Seems kind of like we'd be duplicating that part of `synphot`?

Reply:
Not necessarily, in your pre-processing step before shipping the file with this package, you can write out that data in whatever form specutils can parse.

Reply:
Con: More work for you but you can script it up and also ship the script as part of this package if you want.

Pro: Your package will be much more lightweight without having to worry about PYSYN_CDBS.

Reply:
I'm also reminded that reproducibility is a big quality attribute we're trying to accomplish here (A lot of this project has been trying to track down information from various, desperate sources and we're trying to archive a lot of that digging here). I think these are great ideas; we should save these for discussion when `astrogrism` begins to mature. The code as it's written is probably still good to include just to show *HOW* we generate our synthetic spectrum, even if eventually one day we do as you're suggesting here and just writing out either the bandpass or the full spectrum to another file we simply read from (though now that I say this, a notebook might also suffice, but I think that can be for another day)

Reply:
Well, if you want to keep the code as is, let me follow up on this other comment.

> equivalent checks for the reference atlas

On import, stsynphot really only checks for Vega because it needs to preload Vega for historical reasons. I don't think you have to really check for reference atlas, as `grid_to_spec` would fail anyway if it is calling an atlas that you didn't "install" properly.

Reply:
I came up with a half solution here. Since we're caching the actual HST instrument tarfile from `astropy.download_file`, I'm checking each file within the tarfile and only extracting the ones that are missing. Understandably, this doesn't skip the need to always have that entire tarfile, but at least it doesn't needlessly overwrite files that already exist

Oh, and I also implemented the file check for the vega atlas as well